### PR TITLE
pam_gnupg: init at 0.1

### PR DESCRIPTION
**Pull request by mtreca**
###### Motivation for this change

Add the pam_gnupg module that unlocks gnupg keys on login.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]: No `meta`?

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]: Why not use https://github.com/cruegge/pam-gnupg/releases/tag/v0.1 ?

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]: Shouldn't `pam` and `gnupg` be present in `buildInputs`?

Also, in accordance with above:

```suggestion
  nativeBuildInputs = [ autoreconfHook libtool pam gnupg ];
```

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]:
```suggestion
{stdenv, fetchFromGitHub, autoreconfHook, libtool, pam, gnupg }:
```

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]: After looking at upstraem's `autogen.sh`, `autoreconfHook` should work out of the box. So please remove this and if the only thing left from this is the requested `./configure` command, use:

```
  configureFlags = [
    "--with-moduledir=${placeholder "out"}/lib/security"
  ];
```

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (vxid) [reply pkgs/os-specific/linux/pam_gnupg/default.nix]: Oops. Done.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (vxid) [reply pkgs/os-specific/linux/pam_gnupg/default.nix]: Good point. I just ended up using a release in the new expression since it contains a security fix.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (vxid) [reply pkgs/os-specific/linux/pam_gnupg/default.nix]: Done.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (vxid) [reply pkgs/os-specific/linux/pam_gnupg/default.nix]: Good point. Done.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]: Please use `fetchFromGitHub`

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (doronbehar) [inline pkgs/os-specific/linux/pam_gnupg/default.nix]: Oh and it seems there's a problem with your username :)
```suggestion
    maintainers = with maintainers; [ vxid ];
```

Comment (vxid) [reply pkgs/os-specific/linux/pam_gnupg/default.nix]: My bad. I am about to change Github accounts so I put my new identity there.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (vxid) [reply pkgs/os-specific/linux/pam_gnupg/default.nix]: Done!

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Thanks for your contribution @vxid, one more thing I noticed now: Please change the name of the PR to reflect the commit message + usually most of the times it's `init` not `Init` (lowercase).

**Comment by vxid** [comment]
@doronbehar All done, thanks a lot for your help and feedback.

**Review by doronbehar — APPROVED** [review]

**Comment by mtreca** [comment]
Hm, I still think it would be good to merge this PR. @doronbehar any idea who could be able to merge this? Thanks in advance!

File: maintainers/maintainer-list.nix
Comment (zowoq) [inline maintainers/maintainer-list.nix:59]: https://github.com/NixOS/nixpkgs/blob/e6ec21dfa3feaafccf90a5234d4f0b5c3a48cddb/maintainers/maintainer-list.nix#L29-L34

We now have a rule about handles that begin with a numeral.
```suggestion
  _0x6d6178 = {
```

 Please also rebase the PR on master.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (zowoq) [inline pkgs/os-specific/linux/pam_gnupg/default.nix:28]:
```suggestion
    maintainers = with maintainers; [ _0x6d6178 ];
```

**Review by zowoq — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (zowoq) [inline pkgs/os-specific/linux/pam_gnupg/default.nix:25]: Homepage should be quoted per [RFC45](https://github.com/NixOS/rfcs/pull/45)

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (zowoq) [inline pkgs/os-specific/linux/pam_gnupg/default.nix:3]:
```suggestion

```

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (zowoq) [inline pkgs/os-specific/linux/pam_gnupg/default.nix:24]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (zowoq) [inline pkgs/os-specific/linux/pam_gnupg/default.nix:22]: `autoreconfHook` should include `libtool`.

File: pkgs/os-specific/linux/pam_gnupg/default.nix
Comment (zowoq) [inline pkgs/os-specific/linux/pam_gnupg/default.nix:6]: Not sure why an underscore is used here, upstream has `pam-gnupg`?

**Comment by NickHu** [comment]
@mtreca Thanks for your contribution! I rebased onto master for you and made the minor changes suggested by @zowoq. I would have merged too, but I can't find a GitHub account under the username `0x6d6178` - you mentioned something about switching accounts, so I'll leave it to you to update the maintainer information and then I can merge.

**Comment by mtreca** [comment]
@zowoq Sorry for the long delay, and thanks for the code review.
@NickHu Thanks a lot for implementing these changes. I did change (again) my github username, and updated the files accordingly.

**Pull request merged**
